### PR TITLE
sepolicy: add mostly working sepolicy

### PR DESCRIPTION
**Pull request by Grimler91**
Tested on dream2lte, running in enforcing works fine and I am now only seeing a few denials I have not figured out how to fix yet:

```
kworker/u16:1: type=1400 audit(0.0:85): avc: denied { dac_override } for capability=1 scontext=u:r:kernel:s0 tcontext=u:r:kernel:s0 tclass=capability permissive=0
kworker/u16:1: type=1400 audit(0.0:86): avc: denied { dac_read_search } for capability=2 scontext=u:r:kernel:s0 tcontext=u:r:kernel:s0 tclass=capability permissive=0
type=1400 audit(1598966762.983:15): avc: denied { dac_override } for comm="e2fsck" capability=1 scontext=u:r:fsck:s0 tcontext=u:r:fsck:s0 tclass=capability permissive=0
type=1400 audit(1598966762.983:15): avc: denied { dac_read_search } for comm="e2fsck" capability=2 scontext=u:r:fsck:s0 tcontext=u:r:fsck:s0 tclass=capability permissive=0
```

The policy was created by comparing with the policies of android_device_samsung_universal7880-common, android_device_samsung_exynos9820 and android_device_samsung_universal9810-common (and probably a few others), and with lots of trial and error.

### CODE REVIEW

**Comment by Stricted** [comment]
thanks for the pr, i will look over it as soon as i have time for it

**Comment by Grimler91** [comment]
No stress. 

Force-pushed to remove the call recording commit.

**Comment by Stricted** [comment]
after a quick look:
- remove the untrusted_app*.te rules, they shouldnt be allowed
- alot of things could use macros e.g. r_dir_perms, r_file_perms

**Comment by Grimler91** [comment]
Thanks for the hints!

> alot of things could use macros e.g. r_dir_perms, r_file_perms

How much macros should I use, examples:

I have `allow hal_light_default sysfs_virtual:file { open write getattr };`, is `allow hal_light_default sysfs_virtual:file rw_file_perms;` okay here (it allows `append lock map read ioctl lock map watch watch_reads` in addition to `open write getattr`) or is it too permissive?

I have `allow init sysfs_graphics:file { setattr open read write };`, is `allow init sysfs_graphics:file create_file_perms;` okay here (it allows `create rename unlink getattr ioctl lock map watch watch_reads open append lock map` in addition to `setattr open read write`) or is it too permissive?

**Comment by Stricted** [comment]
in both examples it would be fine to use the macros imo

**Comment by Grimler91** [comment]
@Stricted alright, updated with macros in most files. I have built and tested it without any obvious problems so far.

File: sepolicy/hal_audio_default.te [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+allow hal_audio_default rild:unix_stream_socket connectto;
+allow hal_audio_default system_suspend_hwservice:hwservice_manager find;
+
+# /efs/maxim/rdc_cal
+allow hal_audio_default efs_file:file r_file_perms;
+allow hal_audio_default efs_file:dir search;
+
+allow hal_audio_default imei_efs_file:dir search;
+allow hal_audio_default imei_efs_file:file r_file_perms;
+allow hal_audio_default vendor_radio_prop:file { getattr open read };
```
Comment (Stricted) [inline sepolicy/hal_audio_default.te]: r_file_perms

File: sepolicy/crash_dump.te [hunk 2]
```diff
@@ -0,0 +1,3 @@
+allow crash_dump hwservicemanager_prop:file { getattr open };
```
Comment (Stricted) [inline sepolicy/crash_dump.te]: r_file_perms

File: sepolicy/crash_dump.te [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+allow crash_dump exported_camera_prop:file { getattr open };
```
Comment (Stricted) [inline sepolicy/crash_dump.te]: r_file_perms

File: sepolicy/hal_camera_default.te [hunk 4]
```diff
@@ -0,0 +1,14 @@
+vndbinder_use(hal_camera_default)
+
+allow hal_camera_default vndbinder_device:chr_file r_file_perms;
+allow hal_camera_default hal_graphics_mapper_hwservice:hwservice_manager find;
+allow hal_camera_default hal_graphics_composer_default:fd use;
+allow hal_camera_default sysfs_virtual:dir search;
+allow hal_camera_default sysfs_virtual:file rw_file_perms;
+allow hal_camera_default sysfs_camera:dir search;
+allow hal_camera_default sysfs_camera:file rw_file_perms;
+allow hal_camera_default exported_camera_prop:file { getattr open read };
```
Comment (Stricted) [inline sepolicy/hal_camera_default.te]: r_file_perms

File: sepolicy/hal_keymaster_default.te [hunk 5]
```diff
@@ -0,0 +1 @@
+allow hal_keymaster_default tee_prop:file { getattr open read };
```
Comment (Stricted) [inline sepolicy/hal_keymaster_default.te]: r_file_perms

File: sepolicy/hal_vibrator_default.te [hunk 6]
```diff
@@ -0,0 +1,2 @@
+allow hal_vibrator_default sysfs_virtual:dir search;
+allow hal_vibrator_default sysfs_virtual:file { open write getattr };
```
Comment (Stricted) [inline sepolicy/hal_vibrator_default.te]: rw_file_perms

File: sepolicy/hal_wifi_hostapd_default.te [hunk 7]
```diff
@@ -0,0 +1,2 @@
+allow hal_wifi_hostapd_default sysfs_virtual:dir search;
+allow hal_wifi_hostapd_default sysfs_virtual:lnk_file { read getattr };
```
Comment (Stricted) [inline sepolicy/hal_wifi_hostapd_default.te]: r_file_perms

File: sepolicy/system_server.te [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+allow system_server gpu_device:chr_file rw_file_perms;
+
+# memtrack HAL
+allow system_server debugfs_mali:dir r_dir_perms;
+allow system_server debugfs_mali:file r_file_perms;
+allow system_server debugfs_ion:file r_file_perms;
+allow system_server debugfs_mali_mem:file r_file_perms;
+
+allow system_server frp_block_device:blk_file rw_file_perms;
+allow system_server vendor_radio_prop:file { getattr open read };
```
Comment (Stricted) [inline sepolicy/system_server.te]: r_file_perms

**Review by Stricted — COMMENTED** [review]
some minor things


also would it possible for you to split out the trustzone (tee.te & related) stuff out into https://github.com/LineageOS/android_device_samsung_slsi_sepolicy (tee/mobicore) so we have a generic implementation there for mebicore?

**Comment by Grimler91** [comment]
> also would it possible for you to split out the trustzone (tee.te & related) stuff out into https://github.com/LineageOS/android_device_samsung_slsi_sepolicy (tee/mobicore) so we have a generic implementation there for mebicore?

Yeah, sure. Should I move only mobicore/tee related files/lines or also merge all files that already exist in android_device_samsung_slsi_sepolicy? 
I have several files (hal_*.te, lhd.te, rild.te, ..) that exist in android_device_samsung_slsi_sepolicy as well, should I merge them as well?

**Comment by Stricted** [comment]
i would say just do the mobicore stuff for now

**Comment by Grimler91** [comment]
Have pushed the mobicore stuff to gerrit for review now. 

I can also squash some commits in this PR once everything looks good to keep the git history nicer.

**Comment by Stricted** [comment]
left a comment on gerrit, will test the sepolicy stuff in the next days when i get around to it

**Comment by Stricted** [comment]
i didnt test it myself yet but it looks good as is hence i merged it, will hopefully get around to actually testing it soon

**Pull request merged**